services/replayer/blocks_provider.go

```diff
@@ -38,16 +38,24 @@ func (bs *blockSnapshot) BlockContext() (evmTypes.BlockContext, error) {
 				return gethCommon.Hash{}
 			}
 
-			block, err := bs.blocks.GetByHeight(n)
-			if err != nil {
+			if bs.chainID == flowGo.Emulator {
+				block, err := bs.blocks.GetByHeight(n)
+				if err != nil {
+					return gethCommon.Hash{}
+				}
+				blockHash, err := block.Hash()
+				if err != nil {
+					return gethCommon.Hash{}
+				}
+
+				return blockHash
+			} else if bs.chainID == flowGo.Testnet || bs.chainID == flowGo.Mainnet {
+				// For testnet & mainnet, we fetch the block hash from the hard-coded
+				// array of hashes.
+				return bs.fixedHashes[bs.block.Height%256]
```

Comment:
should be `n % 256`

Reply:
Nice catch :100: Updated in https://github.com/onflow/flow-evm-gateway/pull/659/commits/73f0daa45e55fb69624e82147d711cdb049f6232